flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionManagerTest.java

```diff
@@ -55,6 +68,29 @@ public void testCreateViewForRegisteredPartition() throws Exception {
                 partition.getPartitionId(), 0, new NoOpBufferAvailablityListener());
     }
 
+    @Test
+    public void testCreateViewNotifierAfterRegisteredPartition() throws Exception {
+        final ResultPartitionManager partitionManager = new ResultPartitionManager();
+        final ResultPartition partition = createPartition();
+
+        partitionManager.registerResultPartition(partition);
+        PartitionRequestNotifier partitionRequestNotifier = new TestingPartitionRequestNotifier();
+        assertNotNull(partitionManager.createSubpartitionViewOrNotify(
+                partition.getPartitionId(), 0, new NoOpBufferAvailablityListener(), partitionRequestNotifier));
+    }
+
+    @Test
+    public void testCreateViewNotifierBeforeRegisteredPartition() throws Exception {
+        final ResultPartitionManager partitionManager = new ResultPartitionManager();
+        final ResultPartition partition = createPartition();
+
+        PartitionRequestNotifier partitionRequestNotifier = new TestingPartitionRequestNotifier();
+        assertNull(partitionManager.createSubpartitionViewOrNotify(
+                partition.getPartitionId(), 0, new NoOpBufferAvailablityListener(), partitionRequestNotifier));
+
```

Comment:
instead of `assertNull`, you should:
1. check that list of registered listeners/notifiers in the `ResultPartitionManager` is NOT empty
2. assert that `partitionRequestNotifier` has NOT been notified

Reply:
Done